src/tutorials/nft_cosmetics.md

```diff
@@ -0,0 +1,380 @@
+
+![!Properties](../img/NFTCosmetics/props.png){: .center loading="lazy" }
+
+### NFT Token Data Table
+
+Because the NFTs in the collection are lazily minted, there isn't a way to get all NFTs from a collection due to the contract address being a global address for lazy minted NFTs. This means if you try to retrieve NFTs using the contract address, it may return a random NFT created by someone else. The solution to this would be to put your NFTs on the blockchain, but this costs gas.
+
+Another way we can solve this is by having a data table with the collection of NFT Token IDs, and we randomly pick a Token ID from the data table.
+
+#### Create Cosmetic Tokens Data Table
+
+Create a new data table called `Cosmetic Tokens`. This table will have 1 column called `TokenID` which is a string type.
+
+Add as many tokens from the [collection](https://opensea.io/collection/core-box-cosmetic) by either copying the token ID from the URL or the **Details** section on the NFT page.
+
+![!Cosmetic Tokens](../img/NFTCosmetics/tokens.png){: .center loading="lazy" }
+
+## Create Cosmetics Data Table
+
+We need a data table that will hold the templates that link to each **Box** attribute.
+
+Create a data table called `Cosmetics` and add 1 column called `Template` that is an **Asset Reference** type.
+
+![!Cosmetics](../img/NFTCosmetics/cosmetics.png){: .center loading="lazy" }
+
+### Create Templates
+
+Create some cosmetic templates that will be spawned and attached to the player when they walk through the trigger. For this tutorial, we are using the cardboard boxes from **Core Content**.
+
+![!Box](../img/NFTCosmetics/box.png){: .center loading="lazy" }
+
+### Add Templates to Data Table
+
+When you have created your templates for the boxes, they need to be added to the Cosmetics data table.
+
+![!Rows](../img/NFTCosmetics/cosmetic_rows.png){: .center loading="lazy" }
+
+## Create Trigger
+
+To apply a random cosmetic to the player, you will need to set up an area that has a trigger so that players can enter it and receive a cosmetic. In this case, we will make it so it applies a random one to the player.
+
+1. Create a Client Context.
+2. Create a trigger and scale it up so that players can enter it easily.
+3. Add some geo so players know where to go to use and change their cosmetics.
+4. Turn off the collision on the trigger.
+
+![!Trigger](../img/NFTCosmetics/trigger.png){: .center loading="lazy" }
+
+## Create Loading Bar
+
+NFTs can take a little while to load. Having a progress bar to show the player will be a good idea so they know something is happening.
+
+There are a few ways a loading bar could be done. In this section, we will create a 3D loading bar that will update as NFTs are loaded.
+
+1. Create a group called `Loading`.
+2. Adding 3D text objects that make up the word `Loading`.
+3. Add a bar below the Loading text by setting the **Z** position to `-50`.
```

Comment:
instead of saying "bar", it may be more clear to say Cube- Bottom Aligned (assuming thats what youre using)

Reply:
if its not a cube then I was confused on where to find the "bar" object